src/Avalonia.Controls/Label.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Avalonia.Controls.Primitives;
+using Avalonia.Controls.Templates;
+using Avalonia.Data;
+using Avalonia.Input;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Label control. Focuses <see cref="Target"/> on pointer click or access key press (Alt + accessKey)
+    /// </summary>
+    public class Label : ContentControl
+    {
+        /// <summary>
+        /// Defines the <see cref="Target"/> Direct property
+        /// </summary>
+        public static readonly DirectProperty<Label, IInputElement> TargetProperty =
+            AvaloniaProperty.RegisterDirect<Label, IInputElement>(nameof(Target), lbl => lbl.Target, (lbl, inp) => lbl.Target = inp);
+
+        /// <summary>
+        /// Label focus target storage field
+        /// </summary>
+        private IInputElement _target;
+
+        /// <summary>
+        /// Flag indicating that custom template was provided to show Access Key
+        /// </summary>
+        private bool _isContentTemplateProvided;
+
+        /// <summary>
+        /// Label focus Target
+        /// </summary>
+        public IInputElement Target
+        {
+            get => _target;
+            set => SetAndRaise(TargetProperty, ref _target, value);
+        }
+
+        /// <summary>
+        /// Initializes instance of <see cref="Label"/> control
+        /// </summary>
+        public Label()
+        {
+            this.GetObservable(ContentProperty).Subscribe(ContentChanged);
+        }
+
+        /// <summary>
+        /// Method which focuses <see cref="Target"/> input element
+        /// </summary>
+        private void LabelActivated()
+        {
+            Target?.Focus();
+        }
+
+        /// <summary>
+        /// Handler for Content property change event
+        /// </summary>
+        /// <param name="obj">new value</param>
+        private void ContentChanged(object obj)
```

Comment:
@grokys 
It seems that WPF implements support for access keys directly in ContentPresenter. Should we port that behavior or just keep hacks like this one?

Reply:
> Just curious: Why? Label always seemed like redundant control to me.

It is very needed by users who works with keyboard/hotkeys. I know such users and their work contains of entering many documents with complex structure and ability of using hotkeys to travel through inputs is very important for them. 
They hate programs without hotkeys.